### PR TITLE
[DSIP-39][parameter] Improvement startup parameters/global parameters/project parameters data type

### LINKED ISSUE #15936: [DSIP-39][parameter] Startup parameters/global parameters/project parameters support set data type
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Motivation
Currently startup parameters, global parameters, project parameters do not support custom data value types, the default is VARCHAR data value type, if I want to use the data type is LONG, this time there is no way to carry out the operation, so this optimization point is to give more choice points to the user, to increase the user experience.

### Design Detail
一、 DESC：
This PR is for the startup parameter, global parameter, project parameter support custom data value type, to meet our needs for data use, increase user experience. Which I did not modify the logic of the big, just load the data value type to personalize the limited operation, specifically the following points:
- 1.1 the default data type of startup parameters, global parameters, project parameters is VARCHAR, then this time I transformed this, support for custom data types and compatible with the original data type operation.
- 1.2 StartParams was originally MAP way, now changed to LIST, which I judged whether it is JSON compatible operation.
- 1.3 add New UI
- 1.4 add UT

二、 EFFECT DISPLAY：
2.1 GIF

2.2 Startup Parameters
启动参数
2.3 Global Parameters
全局参数
2.4 Project Parameters
项目参数

### Compatibility, Deprecation, and Migration Plan

Compatibility with latest version.

### Test Plan

Test by UT.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sdhzwc**
close #15936 

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by ruanwenjun — CHANGES_REQUESTED** [review]
Need to discuss at #15936.

**Review by songjianet — DISMISSED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectParameterController.java [hunk 1]
```diff
@@ -61,24 +61,27 @@ public class ProjectParameterController extends BaseController {
...
     })
     @PostMapping()
     @ResponseStatus(HttpStatus.CREATED)
     @ApiException(CREATE_PROJECT_PARAMETER_ERROR)
     public Result createProjectParameter(@Parameter(hidden = true) @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
                                          @Parameter(name = "projectCode", description = "PROJECT_CODE", required = true) @PathVariable long projectCode,
                                          @RequestParam("projectParameterName") String projectParameterName,
-                                         @RequestParam(value = "projectParameterValue") String projectParameterValue) {
+                                         @RequestParam(value = "projectParameterValue") String projectParameterValue,
+                                         @RequestParam(value = "projectParameterDataType") String projectParameterDataType) {
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectParameterController.java]:
```suggestion
                                         @RequestParam(value = "projectParameterDataType", defaultValue = "VARCHAR") String projectParameterDataType) {
```

Comment (sdhzwc) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectParameterController.java]: > 

Modification completed

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ExecutorController.java [hunk 2]
```diff
@@ -163,9 +169,18 @@ public Result startProcessInstance(@Parameter(hidden = true) @RequestAttribute(v
...
+            boolean isJson = jsonElement.isJsonObject();
+            if (isJson) {
+                Map<String, String> startParamMap = JSONUtils.toMap(startParams);
+                startParamList = startParamMap.entrySet().stream()
+                        .map(entry -> new Property(entry.getKey(), Direct.IN, DataType.VARCHAR, entry.getValue()))
+                        .collect(Collectors.toList());
+            } else {
+                startParamList = JSONUtils.toList(startParams, Property.class);
+            }
         }
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ExecutorController.java]: Add a utils to transform the json to Propertity list

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ExecutorController.java [hunk 3]
```diff
@@ -262,9 +277,18 @@ public Result batchStartProcessInstance(@Parameter(hidden = true) @RequestAttrib
...
+            JsonElement jsonElement = JsonParser.parseString(startParams);
+            boolean isJson = jsonElement.isJsonObject();
+            if (isJson) {
+                Map<String, String> startParamMap = JSONUtils.toMap(startParams);
+                startParamList = startParamMap.entrySet().stream()
+                        .map(entry -> new Property(entry.getKey(), Direct.IN, DataType.VARCHAR, entry.getValue()))
+                        .collect(Collectors.toList());
+            } else {
+                startParamList = JSONUtils.toList(startParams, Property.class);
+            }
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ExecutorController.java]: same

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java [hunk 4]
```diff
@@ -760,8 +761,8 @@ private int createCommand(Long triggerCode, CommandType commandType, long proces
         if (warningType != null) {
             command.setWarningType(warningType);
         }
-        if (startParams != null && startParams.size() > 0) {
-            cmdParam.put(CMD_PARAM_START_PARAMS, JSONUtils.toJsonString(startParams));
+        if (startParamList != null && startParamList.size() > 0) {
+            cmdParam.put(CMD_PARAM_START_PARAMS, JSONUtils.toJsonString(startParamList));
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]: Use `CollectionUtils.isNotEmpty(startParamList)`

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

**Comment by ruanwenjun** [comment]
Frontend compile failed.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java (hunk 4 above)
Comment (sdhzwc) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]: > Use `CollectionUtils.isNotEmpty(startParamList)`

done

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ExecutorController.java (hunk 2 above)
Comment (sdhzwc) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ExecutorController.java]: > Add a utils to transform the json to Propertity list

done

**Comment by sdhzwc** [comment]
> Frontend compile failed.

done

**Comment by ruanwenjun** [comment]
Sorry I push to your branch, but the pr auto closed, please reopen it.

**Comment by sdhzwc** [comment]
Start a Review

**Pull request closed without merging**